helper/broker/notify_test.go

```diff
@@ -23,36 +23,65 @@ func TestGenericNotifier(t *testing.T) {
 	go notifier.Run(stopChan)
 
 	// Ensure we have buffered channels.
-	require.Equal(t, 1, cap(notifier.publishCh))
-	require.Equal(t, 1, cap(notifier.subscribeCh))
-	require.Equal(t, 1, cap(notifier.unsubscribeCh))
+	must.Eq(t, 1, cap(notifier.publishCh))
+	must.Eq(t, 1, cap(notifier.subscribeCh))
+	must.Eq(t, 1, cap(notifier.unsubscribeCh))
 
 	// Test that the timeout works.
 	var timeoutWG sync.WaitGroup
 
 	for i := 0; i < 6; i++ {
 		go func(wg *sync.WaitGroup) {
 			wg.Add(1)
+			defer wg.Done()
+
 			msg := notifier.WaitForChange(100 * time.Millisecond)
-			require.Equal(t, "wait timed out after 100ms", msg)
-			wg.Done()
+			must.Eq(t, "wait timed out after 100ms", msg)
 		}(&timeoutWG)
 	}
 	timeoutWG.Wait()
 
-	// Test that all subscribers recieve an update when a single notification
+	// Test that all subscribers receive an update when a single notification
 	// is sent.
-	var notifiedWG sync.WaitGroup
+	resultCh := make(chan interface{})
+	startedCh := make(chan struct{})
 
 	for i := 0; i < 6; i++ {
-		go func(wg *sync.WaitGroup) {
-			wg.Add(1)
-			msg := notifier.WaitForChange(3 * time.Second)
-			require.Equal(t, "we got an update and not a timeout", msg)
-			wg.Done()
-		}(&notifiedWG)
+		go func() {
+			select {
+			default:
+				startedCh <- struct{}{}
+			}
+			resultCh <- notifier.WaitForChange(3 * time.Second)
+		}()
 	}
 
+	// Wait until all routines have started and fallen-through to waiting for
+	// the change to occur.
+	var started int
+
+	select {
+	case <-startedCh:
+		started++
+	default:
+		if started == 6 {
+			break
+		}
+	}
```

Comment:
This `select` isn't in a loop - isn't it going to block just once and `started` will end up always == 1?